src/utils/Mappers/PermissionMapper.res

```diff
@@ -0,0 +1,99 @@
+  | UnknownPermission(val) => val
+  }
+}
+
+let mapStringToPermissionType = val => {
+  switch val {
+  | "operations_view" => OperationsView
+  | "operations_manage" => OperationsManage
+  | "connectors_view" => ConnectorsView
+  | "connectors_manage" => ConnectorsManage
+  | "workflows_view" => WorkflowsView
+  | "workflows_manage" => WorkflowsManage
+  | "analytics_view" => AnalyticsView
+  | "users_view" => UsersView
+  | "users_manage" => UsersManage
+  | "merchant_details_view" => MerchantDetailsView
+  | "merchant_details_manage" => MerchantDetailsManage
+  | "organization_manage" => OrganizationManage
+  | val => UnknownPermission(val)
+  }
+}
+
+let defaultValueForPermission = {
+  operationsView: NoAccess,
+  operationsManage: NoAccess,
+  connectorsView: NoAccess,
+  connectorsManage: NoAccess,
+  workflowsView: NoAccess,
+  workflowsManage: NoAccess,
+  analyticsView: NoAccess,
+  usersView: NoAccess,
+  usersManage: NoAccess,
+  merchantDetailsView: NoAccess,
+  merchantDetailsManage: NoAccess,
+  organizationManage: NoAccess,
+}
+
+let hasAnyPermission = (permission1, permission2) => {
+  switch (permission1, permission2) {
+  | (NoAccess, NoAccess) => NoAccess
+  | (_, _) => Access
+  }
+}
+
+let getAccessValue = (~permissionValue: permissionType, ~permissionList) => {
+  let isPermissionFound = permissionList->Array.find(ele => {
+    ele == permissionValue
+  })
+
+  isPermissionFound->Option.isSome ? Access : NoAccess
+}
+
+// TODO: Refactor to not call function for every permission
+let getPermissionJson = permissionList => {
+  {
+    operationsView: getAccessValue(~permissionValue=OperationsView, ~permissionList),
+    operationsManage: getAccessValue(~permissionValue=OperationsManage, ~permissionList),
+    connectorsView: getAccessValue(~permissionValue=ConnectorsView, ~permissionList),
+    connectorsManage: getAccessValue(~permissionValue=ConnectorsManage, ~permissionList),
+    workflowsView: getAccessValue(~permissionValue=WorkflowsView, ~permissionList),
+    workflowsManage: getAccessValue(~permissionValue=WorkflowsManage, ~permissionList),
+    analyticsView: getAccessValue(~permissionValue=AnalyticsView, ~permissionList),
+    usersView: getAccessValue(~permissionValue=UsersView, ~permissionList),
+    usersManage: getAccessValue(~permissionValue=UsersManage, ~permissionList),
+    merchantDetailsView: getAccessValue(~permissionValue=MerchantDetailsView, ~permissionList),
+    merchantDetailsManage: getAccessValue(~permissionValue=MerchantDetailsManage, ~permissionList),
+    organizationManage: getAccessValue(~permissionValue=OrganizationManage, ~permissionList),
+  }
+}
+
+let convertValueToMap = arrayValue => {
+  open CommonAuthTypes
+  Js.log2("arrayValue", arrayValue)
+  let userPermissionmap: Map.t<
+    UserManagementTypes.permissionType,
+    CommonAuthTypes.authorization,
+  > = Map.make()
+  arrayValue->Array.forEach(value => {
+    userPermissionmap->Map.set(value, Access)
```

Comment:
setting this to Access by default because the array we're iterating over currently only contains modules the user has access to.